docs/core/extensions/snippets/logging/di-without-host/di-without-host.csproj

```diff
@@ -0,0 +1,17 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net7.0</TargetFramework>
```

Comment:
What is our usual stance on using new versions of .NET in samples prior to that version being released? The docs don't currently mention .NET 8 is a pre-requisite or use any features that are specific to .NET 8. 

For primary constructors my personal opinion would be that we should be conservative in the language features we use in samples. Many devs may be unfamiliar with the new features and then it would be harder for them to understand the sample content if those features are used. Of course if you tell me the docs team has considered this in the past and found it isn't an issue I'll trust your expertise.

Reply:
That is a fair question. We are slowing adding .NET 8 bits into the fold, since we're so close to .NET 8 going GA - I'd advocate for that. And for introducing new C# language features into content, we strive to do this as much as possible. For the very reasons that you mentioned, many devs are unfamiliar with them and we believe this is the appropriate place to show them how to use these features. The .NET docs team is consistently introducing the new C# syntax throughout our examples.

Reply:
I added a primary constructor the first place constructor injection comes up. There are other potential places it could be used but I didn't want to get that aggressive with it.